### PR TITLE
Fixes and improvements to N52xx PNA Driver

### LINKED ISSUE #3687: N52xx PNA Driver Warnings on Initialization
On initialization, drivers based on the N52xx PNA base class throw warnings about **kwargs not being forwarded on traces.

### Steps to reproduce
1. Initialize a driver based on the N52xx base class (e.g. `qcodes.instrument_drivers.Keysight.N5245A.N5245A`)

### Actual behaviour
Warning thrown:
```
2021-12-15 17:10:26,061 ¦ py.warnings ¦ WARNING ¦ warnings ¦ _showwarnmsg ¦ 109 ¦ c:\users\xxxxx\qcodes\qcodes\qcodes\instrument\base.py:129: QCoDeSDeprecationWarning: Parameter imaginary on instrument pna_tr4 does not correctly pass kwargs to its baseclass. A Parameter class must take `**kwargs` and forward them to its baseclass.
```

### PR DESCRIPTION
**Pull request by spauka**
Changes proposed in this PR:
 - Fix warnings on instrument initialization caused by **kwargs not being forwarded by `PNASweep` (closes #3687)
 - Use new instrument logger
 - Update old `ArrayParameters` to use `ParameterWithSetpoints` instead
 - Correctly label axis labels when sweeping in CW mode or with a LOG frequency sweep

~~Note: This PR depends on #3683. I will rebase it once that changeset is merged.~~

@jenshnielsen

### CODE REVIEW

File: docs/changes/newsfragments/3688.n52xx [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- Update N52xx driver to fix warnings on initialization
```
Comment (astafan8) [inline docs/changes/newsfragments/3688.n52xx:1]: this file should be called 3688.improved_driver :) see here https://qcodes.github.io/Qcodes/community/contributing.html#pull-requests

**Review by astafan8 — APPROVED** [review]
thank you @spauka !

File: docs/changes/newsfragments/3688.n52xx (hunk 1 above)
Comment (spauka) [reply docs/changes/newsfragments/3688.n52xx]: Thanks, noted and fixed :)

**Comment by spauka** [comment]
Since 3683 is now merged I've rebased this PR and it should be ready to merge.

I've tested it with the N5245A network analyzer and it works correctly.

**Comment by spauka** [comment]
@astafan8 I'm not sure I understand why the docs are failing to build?

File: docs/changes/newsfragments/3688.improved_driver [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Update N52xx driver to fix warnings on initialization
+related to not passing **kwargs in trace parameters (closes #3687).
```
Comment (jenshnielsen) [inline docs/changes/newsfragments/3688.improved_driver:2]: @spauka This should be valid sphinx rst. In rst `**` starts a bold section (Inline strong) this never ends here. I suggest that we wrap this in double ticks ``` ``**kwargs`` ```

Comment (spauka) [reply docs/changes/newsfragments/3688.improved_driver]: Thanks for the tip - that seems to have fixed it :)

**Comment by jenshnielsen** [comment]
@spauka In general you don't need to worry about adding prama comments to disable codacy warnings. We look at codacy as guidelines not as requirements, which is also why the job is not marked as required to pass

**Comment by spauka** [comment]
Ah, thanks. I really need to re-read the contributions guide, I just noticed it was there too 😆 

Would you like me to remove the codacy comments? The mypy ones I think have to stay otherwise the CI tests fail...

**Comment by jenshnielsen** [comment]
@spauka I don't think they are harmful so feel free to leave them. The mypy ones should stay. For bonus points you can annotate the mypy ones with error type like here https://github.com/QCoDeS/Qcodes/blob/master/qcodes/instrument/channel.py#L880
The easiest way to get these error codes is to run mypy locally with `` --show-error-code`` and without the ignore comments

**Review by astafan8 — APPROVED** [review]
bors merge

**Comment by astafan8** [comment]
bors merge

**Comment by astafan8** [comment]
bors merge

**Pull request merged**